pygfx/objects/_base.py

```diff
@@ -389,152 +379,52 @@ def iter(self, filter_fn=None, skip_invisible=False):
+            return own_sequence
+        else:
+            return self.parent.transform_sequence @ own_sequence
 
     @property
-    def matrix_world(self):
-        """The world matrix (local matrix composed with any parent matrices)."""
-        return self._matrix_world
+    def bounding_box(self):
+        partial_aabb = np.zeros((len(self.children) + 1, 2, 3), dtype=float)
+        for idx, child in enumerate(self.children):
+            aabb = child.bounding_box
+            trafo = child.transform.matrix
+            partial_aabb[idx] = pla.aabb_transform(aabb, trafo)
+        partial_aabb[-1] = self.geometry.bounding_box()
+
+        final_aabb = np.zeros((2, 3), dtype=float)
+        final_aabb[0] = np.min(partial_aabb[:, 0, :], axis=0)
+        final_aabb[1] = np.max(partial_aabb[:, 1, :], axis=0)
+        return final_aabb
 
     @property
-    def matrix_auto_update(self):
-        """Whether or not the matrix auto-updates."""
-        return self._matrix_auto_update
-
-    @matrix_auto_update.setter
-    def matrix_auto_update(self, value):
-        self._matrix_auto_update = bool(value)
+    def bounding_sphere(self):
+        return pla.aabb_to_sphere(self.bounding_box)
 
     @property
-    def matrix_world_dirty(self):
-        """Whether or not the matrix needs updating (readonly)."""
-        return self._matrix_world_dirty
-
-    def apply_matrix(self, matrix):
-        if self._matrix_auto_update:
-            self.update_matrix()
-        self._matrix.premultiply(matrix)
-        self._matrix.decompose(self.position, self.rotation, self.scale)
-        self._matrix_world_dirty = True
-
-    def update_matrix_world(
-        self, force=False, update_children=True, update_parents=False
-    ):
-        if update_parents and self.parent:
-            self.parent.update_matrix_world(
-                force=force, update_children=False, update_parents=True
-            )
-        if self._matrix_auto_update:
-            self.update_matrix()
-        if self._matrix_world_dirty or force:
-            if self.parent is None:
-                self._matrix_world.copy(self._matrix)
-            else:
-                self._matrix_world.multiply_matrices(
-                    self.parent._matrix_world, self._matrix
-                )
-            self.uniform_buffer.data[
-                "world_transform"
-            ].flat = self._matrix_world.elements
-            tmp_inv_matrix = Matrix4().get_inverse(self._matrix_world)
-            self.uniform_buffer.data[
-                "world_transform_inv"
-            ].flat = tmp_inv_matrix.elements
-            self.uniform_buffer.update_range(0, 1)
-            self._matrix_world_dirty = False
-            for child in self._children:
-                child._matrix_world_dirty = True
-        if update_children:
-            for child in self._children:
-                child.update_matrix_world()
-
-    def look_at(self, target: Vector3, *, up=None):
```

Comment:
The `look_at` method and the `up` vector will need to be revived, e.g. see the `look_at.py` example. Probably on your radar, but marking just in case :)

Reply:
Yes. It used to live on `Light` and `Camera` respectively, but I moved it back to `WorldObject` today. The metod itself has some quirk in it (it doesn't rotate in the way we expect), but it is definitely on my radar.